### PR TITLE
Layer menu consistency

### PR DESCRIPTION
**Pull request by roemhildtg**
- Minor changes to the css/html to enable hamburger menus on each sublayer menu. 
 - Implement layer menus on ALL layer types by moving layer menu logic primarily to `_Control`
 - Use separate properties `menu` and `subLayerMenu` for the different menu types 

https://github.com/cmv/cmv-app/issues/603

### Menu 

image

A primary menu for each layer. This can do things like zoom to the layer, toggle all dynamic layers, and get layer metadata.

Note: For dynamic layers with only one layer, this menu is used for the sublayer menu. 

#### Sublayer Menu

image

A menu for layers with sublayers, like dynamic layers 

Usage Examples: 

```javascript
// in layerControl options: 
options: {
    menu: {
        dynamic: [{
                    topic: 'hello',
                    label: 'Say Hello Custom',
                    iconClass: 'fa fa-smile-o'
                }],
        feature: [{
                    topic: 'hello',
                    label: 'Say Hello Custom',
                    iconClass: 'fa fa-smile-o'
                }]
    },
    subLayerMenu: {
        dynamic: [{
                    topic: 'hello',
                    label: 'Say Hello Custom',
                    iconClass: 'fa fa-smile-o'
                }]
        // other types of sublayer menus?
    }
}
```

```javascript
// in layerControlLayerInfos for a specific layer 
layerControlLayerInfos: {
    menu: [{
                    topic: 'hello',
                    label: 'Say Hello Custom',
                    iconClass: 'fa fa-smile-o'
                }],
    subLayerMenu: [{
                    topic: 'hello',
                    label: 'Say Hello Custom',
                    iconClass: 'fa fa-smile-o'
                }]
}
```

### CODE REVIEW

**Comment by tmcgee** [comment]
What do you think of moving the sub-layer menu icons to be right-aligned with a slight indent? Something like this:

image

**Comment by roemhildtg** [comment]
The slight indent bothers me I guess. I would prefer they are aligned as opposed to slightly offset. 

Would the entire text block still be clickable?

File: viewer/js/gis/dijit/LayerControl/controls/templates/Sublayer.html [hunk 1]
```diff
@@ -8,8 +8,8 @@
                 <i data-dojo-attach-point="checkNode" class="fa ${icons.unchecked} layerControlIcon"></i>
             </td>
             <td class="layerControlTableLabel">
-                <span data-dojo-attach-point="labelNode"></span>
-                <span data-dojo-attach-point="iconNode"></span>
+                <span data-dojo-attach-point="labelNode" class="label"></span>
+                <span data-dojo-attach-point="iconNode" class="icon-extra"></span>
```
Comment (tmcgee) [inline viewer/js/gis/dijit/LayerControl/controls/templates/Sublayer.html]: This isn't new in this PR but what is the purpose of `iconNode`? Is it used for anything with sub layers? Could it be used for the menu icon css?

Comment (roemhildtg) [inline viewer/js/gis/dijit/LayerControl/controls/templates/Sublayer.html]: Its so you can do something like this:
```javascript
topic.subscribe('layerControl/menuItemTopic', function (r) {

    //modify the iconNode to show that something happened to this layer
    if (r.iconNode) {
        if (domClass.contains(r.iconNode, 'fa-fire')) {
            domClass.remove(r.iconNode, 'fa-fire');
        } else {
            domClass.add(r.iconNode, 'fa-fire');
        }
    }
});
```

**Comment by tmcgee** [comment]
I was trying to differentiate between the layer and sublayer menus. Removing the indent is ok too. The ragged indent when the icon is left aligned bothers my eyes more than a solid line of layer/sub-layer icons down the right-hand side.

**Comment by roemhildtg** [comment]
What if it gets right aligned 

 + v   = layer name

File: viewer/js/gis/dijit/LayerControl/controls/templates/Sublayer.html (hunk 1 above)
Comment (tmcgee) [inline viewer/js/gis/dijit/LayerControl/controls/templates/Sublayer.html]: ok. Hadn't see it used. Perhaps that should be to the left of a right-aligned menu for consistency with the placement of the spinner next to the layer menu.

**Comment by tmcgee** [comment]
Possibly. If we do that, the layer menu should go there too?

**Comment by roemhildtg** [comment]
Yes, that would be good. And make the whole layer title clickable?

**Comment by tmcgee** [comment]
yep

**Comment by roemhildtg** [comment]
image

**Comment by tmcgee** [comment]
Almost there. 2 things:

1. The margins around the menus are inconsistent between the layer menu and the sublayer menu. the one for the layer needs more space on left/right. 

2. Perhaps the 3 square icons in a row - expand, checkbox, menu are too much. I prefer carets for the expand/collapse. I think that helps.
image

**Comment by roemhildtg** [comment]
Now I'm thinking we should keep the menu's on the right. I still like the caret though. Trying to come up with a way to make the whole layer block clickable so that it expands the layer.

**Comment by DavidSpriggs** [comment]
The burger menu on the right makes the most sense from a UX perspective. Users are used to this and know what it means. When it's in-line with the title, it seems to be confusing and cramped. Perhaps highlighting the row with a slight background color when the cursor hovers on the burger menu would help users identify the correct layer and provide a visual link. (Note that I do not mean when hovering on the entire row, but only when hovering on the burger menu icon itself). Thoughts?

**Comment by roemhildtg** [comment]
replaced with https://github.com/cmv/cmv-app/pull/680

**Pull request closed without merging**